### PR TITLE
Warnings suppression

### PR DESCRIPTION
**Pull request by bartgol**
Suppress a config warning due to cmake policy 0074, as well as a handful of build warnings from f90 (unused vars).

### CODE REVIEW

File: components/cam/src/physics/cam/micro_p3_utils.F90 [hunk 1]
```diff
@@ -85,8 +85,8 @@ module micro_p3_utils
 real(rtype), parameter :: dsph = 3._rtype
 
 ! Bounds for mean diameter for different constituents.
-real(rtype), parameter :: lam_bnd_rain(2) = 1._rtype/[500.e-6_rtype, 20.e-6_rtype]
-real(rtype), parameter :: lam_bnd_snow(2) = 1._rtype/[2000.e-6_rtype, 10.e-6_rtype]
+! real(rtype), parameter :: lam_bnd_rain(2) = 1._rtype/[500.e-6_rtype, 20.e-6_rtype]
```
Comment (jgfouca) [inline components/cam/src/physics/cam/micro_p3_utils.F90]: Can we just remove these lines?

Comment (bartgol) [reply components/cam/src/physics/cam/micro_p3_utils.F90]: I'm ok with that. I just left them there assuming upcoming p3 stuff would need them, so people would only have to uncomment them. @AaronDonahue , thoughts?

Comment (jgfouca) [reply components/cam/src/physics/cam/micro_p3_utils.F90]: I'm OK keeping them if you think they are useful.

Comment (bartgol) [reply components/cam/src/physics/cam/micro_p3_utils.F90]: I don't know if they are. Hopefully Aaron can shed some light.

**Pull request merged**